src/dss/remove_notebook.py

```diff
@@ -0,0 +1,32 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Service
+
+from dss.config import DSS_NAMESPACE
+from dss.logger import setup_logger
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def remove_notebook(name: str, lightkube_client: Client) -> None:
+    """
+    Removes a Notebook server from the Kubernetes cluster.
+
+    Args:
+        name (str): The name of the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+
+    try:
+        lightkube_client.delete(res=Deployment, name=name, namespace=DSS_NAMESPACE)
+        lightkube_client.delete(res=Service, name=name, namespace=DSS_NAMESPACE)
```

Comment:
What about the case where only the service exists?  In that case we try to delete the Deployment, get a 404 error, and exit without trying to delete the Service. 

I propose we refactor this to ensure we try to delete every object, even if a previous one failed.  We do this same approach [in charms](https://github.com/canonical/charmed-kubeflow-chisme/blob/8a7454fede8dbcefe65cd45f2fc04a3926b7e421/src/charmed_kubeflow_chisme/lightkube/batch/_many.py#L78).

Reply:
+1 on this

Reply:
makes sense, I've refactored accordingly